src/Rules/ArrayOfRule.php

```diff
@@ -129,16 +129,12 @@ public function processValue($value, Args $args, FieldContext $context): array
 			}
 
 			if ($itemException !== null || $keyException !== null) {
-				$type->addInvalidPair(
-					$key,
-					$keyException !== null ? $keyException->getInvalidType() : null,
-					$itemException !== null ? $itemException->getInvalidType() : null,
-				);
+				$type->addInvalidPair($key, $keyException, $itemException);
 			}
 		}
 
 		if ($type->hasInvalidParameters() || $type->hasInvalidPairs()) {
-			throw ValueDoesNotMatch::create($type);
+			throw ValueDoesNotMatch::create($type, $value);
 		}
```

Comment:
In case only items are invalid there is no need to track whole value
```php
		if (($invalidParams = $type->hasInvalidParameters()) || $type->hasInvalidItems()) {
			throw ValueDoesNotMatch::create(
				$type,
				!$invalidParams ? NoValue::create() : $value,
			);
		}
```

Reply:
TODO - test case